From ea50a1210c1cb6548e156f74c6fd509811254b91 Mon Sep 17 00:00:00 2001
From: Roy Li <rongqing.li@windriver.com>
Date: Thu, 31 Jan 2019 14:27:58 +0800
Subject: [PATCH] refpolicy: avahi-daemon dontaudit r/w to /dev/console,
 /dev/pts/*

Upstream-Status: Pending

avahi-daemon does not need to read/write /dev/console, so dontaudit
the AVC messages

type=AVC msg=audit(1399622624.430:7): avc: denied { read write } for pid=337 comm="avahi-daemon" path="/dev/console" dev="devtmpfs" ino=5247 scontext=system_u:system_r:avahi_t:s0-s15:c0.c1023 tcontext=root:object_r:user_tty_device_t:s0 tclass=chr_file

Signed-off-by: Roy Li <rongqing.li@windriver.com>
---
 policy/modules/services/avahi.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/services/avahi.te b/policy/modules/services/avahi.te
index c902082..8ef5785 100644
--- a/policy/modules/services/avahi.te
+++ b/policy/modules/services/avahi.te
@@ -78,6 +78,7 @@ fs_list_inotifyfs(avahi_t)
 domain_use_interactive_fds(avahi_t)
 
 files_read_etc_runtime_files(avahi_t)
+files_map_etc_files(avahi_t)
 files_read_usr_files(avahi_t)
 
 auth_use_nsswitch(avahi_t)
@@ -94,6 +95,7 @@ sysnet_domtrans_ifconfig(avahi_t)
 sysnet_manage_config(avahi_t)
 sysnet_etc_filetrans_config(avahi_t)
 
+userdom_dontaudit_use_user_terminals(avahi_t)
 userdom_dontaudit_use_unpriv_user_fds(avahi_t)
 userdom_dontaudit_search_user_home_dirs(avahi_t)
 
-- 
2.7.4

